hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java

```diff
@@ -481,6 +482,25 @@ public void onAlterPartition(AlterPartitionEvent partitionEvent) throws MetaExce
     process(event, partitionEvent);
   }
 
+  @Override
+  public void onAlterPartitions(AlterPartitionsEvent event) throws MetaException {
+    Table table = event.getTable();
+    Iterator<List<Partition>> iterator = event.getNewPartsIterator(maxBatchSize);
```

Comment:
> The JDBC max batch size may not be equal to Thrift max message size. So the consumers of these events may get thrift max message limit error if stats are stored in each of these partitions.

Right, and I don't want to introduce a new property for the batch size. An idea is try to exclude the stats from the message, but I'm not sure if it will hurt the query performance for downstream project using the serialized partitions(without stats), they need to recompute/refetch the stats again